src/lib/common/sol-platform-impl-linux-micro.c

```diff
@@ -514,16 +521,108 @@ teardown_pid1(void)
     reboot(reboot_cmd);
 }
 
+static bool
+sol_platform_linux_micro_should_debug(struct gdbserver *gdb)
+{
+    int idx, argc = sol_argc();
+    char **argv = sol_argv();
+    bool res = false;
+
+    for (idx = 1; idx < argc; idx++) {
```

Comment:
are you sure this works for PID1? I guess we're called without argvs... in that case we'd need to get it from kcmdline.